sdk/servicebus/service-bus/samples/typescript/src/advanced/listingEntities.ts

```diff
@@ -0,0 +1,123 @@
+/*
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT Licence.
+
+  **NOTE**: This sample uses the preview of the next version of the @azure/service-bus package.
+  For samples using the current stable version of the package, please use the link below:
+  https://github.com/Azure/azure-sdk-for-js/tree/%40azure/service-bus_1.1.5/sdk/servicebus/service-bus/samples
+  
+  This sample demonstrates how the ServiceBusManagementClient can be used to list the entities of a service bus namespace.
+
+  See https://docs.microsoft.com/en-us/rest/api/servicebus/resource-provider-apis to learn more.
+*/
+
+import { ServiceBusManagementClient } from "@azure/service-bus";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Define connection string and related Service Bus entity names here
+const connectionString = process.env.SERVICE_BUS_CONNECTION_STRING || "<connection string>";
+
+export async function main() {
+  const serviceBusManagementClient = new ServiceBusManagementClient(connectionString);
+  const baseQueueName = "random-queue";
+  const numberOfQueues = 7;
+
+  // Create queues in order to list them later
+  for (let i = 0; i < numberOfQueues; i++) {
+    await serviceBusManagementClient.createQueue(baseQueueName + "_" + i);
+  }
+
+  // This sample leverages `getQueues()` as an example, you can iterate over topics, subscriptions, and rules
+  // as well as on the runtime info of entities by using methods like `getQueuesRuntimeInfo()`
+  // 1. List Queues
+  let i = 1;
+  let iter = serviceBusManagementClient.getQueues();
+  for await (const queue of iter) {
+    console.log(`Queue ${i++}: ${queue.name}`);
+  }
+
+  // 2. Same as the previous example
+  i = 1;
+  for await (const queue of serviceBusManagementClient.getQueues()) {
+    console.log(`Queue ${i++}: ${queue.name}`);
+  }
+
+  // 3. Generator syntax .next()
+  i = 1;
+  iter = serviceBusManagementClient.getQueues();
+  let queueItem = await iter.next();
```

Comment:
We're walking a line here between showing them how the Azure feature works and just showing them how iterables work. Maybe we should just stick with the one `for await (const queue of <call>)` and not bother with the others?

Reply:
When the async iterator support was added to storage, Brian suggested(or he liked) the idea of having more variations in the examples since these are relatively new to the language and people might get confused.

This is coherent to what I had added for storage long ago.
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-file-share/samples/typescript/src/iterators-shares.ts

Reply:
Makes sense - I figured it was something along those lines. Sounds good to leave as is.

Reply:
At some point we could probably move the details into a common documentation page, but no objections to briefly showing how iterators work.